eth_tester/backends/pyevm/main.py

```diff
@@ -304,7 +304,7 @@ def revert_to_snapshot(self, snapshot):
         block = self.chain.get_block_by_hash(snapshot)
         chaindb = self.chain.chaindb
 
-        chaindb._set_as_canonical_chain_head(chaindb.db, block.header.hash, GENESIS_PARENT_HASH)
+        chaindb._set_as_canonical_chain_head(chaindb.db, block.header, GENESIS_PARENT_HASH)
```

Comment:
wondering if we couldn't use the `eth.tools.builder` suite of tools to eliminate the use of private APIs here.  Instead of using this `snapshot` approach, we could just use the `eth.tools.builder.chain_split` API to split the chain each time we want mark a point for reversion... When we want to do the revert we would just use the forked chain and discard the main chain.

Reply:
Yeah, I can dig that.  Though it will take a bit of re-engineering, right?  Since `chain_sp[lit` wants to execute the callable itself, it will need to go... I'm not sure, somewhere down the stack.   Is `eth.tools.builder.copy` closer to a drop-in replacement?  Of course, we might actually *want* to do things more in the `chain_split` way than the "snapshot" way to begin with.

Anywho, are we good to go on a merge for this one for the moment while we work out the snapshot logic?